File: spring-rabbitmq/src/main/java/com/example/spring_rabbitmq/Sender.java
package com.example.spring_rabbitmq;

import org.springframework.amqp.rabbit.core.RabbitTemplate;

import org.springframework.stereotype.Component;

@Component
public class Sender {
    private final RabbitTemplate rabbitTemplate;
    private static final String SERVICE2_QUEUE = "service2Queue";
    private static final String SERVICE1_QUEUE = "service1Queue";

    public Sender(RabbitTemplate rabbitTemplate) {
        this.rabbitTemplate = rabbitTemplate;
    }


    public void startPingLoop() {
        sendPingToService2();
    }

    public void sendPingToService2() {
        String message = "ping";
        rabbitTemplate.convertAndSend(SERVICE2_QUEUE, message);
        System.out.println("[Service-1] Sent to Service-2: " + message);
    }

    public void sendPongToService1() {
        String message = "pong";
        rabbitTemplate.convertAndSend(SERVICE1_QUEUE, message);
        System.out.println("[Service-1] Sent to service1Queue: " + message);
    }
}

File: SpringKotlin/src/main/kotlin/com/example/SpringKotlin/Sender.kt
package com.example.SpringKotlin

import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
open class Sender @Autowired constructor(private val rabbitTemplate: RabbitTemplate) {

    // Send "pong" to service2Queue
    fun sendPongToService2() {
        rabbitTemplate.convertAndSend("service2Queue", "pong")
        println("[Service-2] Sent pong ")
    }

    // Send "ping" to service1Queue
    fun sendPingToService1() {
        rabbitTemplate.convertAndSend("service1Queue", "ping")
        println("[Service-2] Sent ping to Service-1")
    }
}
